packages/components/CHANGELOG.md

```diff
@@ -17,6 +17,7 @@
 
 ### Enhancements
 
+-   Add `onClick` event on `FormFileUpload` to emit `onChange` event when selecting the same file again.
```

Comment:
I got a bit confused when reading this CHANGELOG entry:

- `onClick` is a property, not an event (at most, it's an event listener)
- we didn't really add it in this PR — prior to this PR, you would already be able to pass `onClick` and it would have been forwarded to the `Button` component
- I would in any case omit the "to emit `onChange` event when selecting the same file again" part — the `onClick` prop can be used for any purpose, not just to allow re-selecting the same file again

As of now, it seems to me that what this PR does is to _explicitly forward the `onClick` prop to the underlying `input` element, rather than the proxy `Button` trigger_

Reply:
* I wanted to say "onClick event listener", it's right, I'll change to property.

* I agree about the `onClick` being already used by forwarding to the `Button` component, so this prop should be `onInputFileClick` as I implemented firstly?

Reply:
> Wouldn't this PR introduce a breaking change?

My thinking was that it wasn't breaking, because if anyone had actually been passing an `onClick`, the component wouldn't work at all unless they did some hacky internal querying in the click handler (`event.target.closest('.components-form-file-upload').querySelector('input[type="file"]').click()` or somesuch). And that kind of breakage isn't covered by our contract. What do you think?

Reply:
> if anyone had actually been passing an onClick, the component wouldn't work at all unless they did some hacky internal querying in the click handler

This is also a fair point. Effectively, with the changes from this PR:

- when the `Button` is clicked, the `input` gets programmatically clicked
- which means that the `onClick` prop should also get called

So, in a way, the `onClick` prop is still called when the `Button` is clicked.

---

In the light of the above:
- let's not label this as a breaking change
- let's still clarify the CHANGELOG entry message (as suggested [above](https://github.com/WordPress/gutenberg/pull/39268#discussion_r830291760))
- let's explicitly list the `onClick` prop in the README